File: Back-end/app/src/main/java/app/entity/Rental.java
package app.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import jakarta.persistence.*;
import lombok.Getter;
import lombok.Setter;
import java.time.LocalDateTime;

@Entity
@Getter
@Setter
public class Rental {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "car_id")
    @JsonBackReference("car-rentals")
    private Car car;

    @ManyToOne
    @JoinColumn(name = "landlord_id")
    @JsonBackReference("landlord-rentals")
    private Landlords landlord;

    @Column(name = "start_date")
    private LocalDateTime startDate;

    @Column(name = "return_date")
    private LocalDateTime returnDate;

    @Column(name = "total_value")
    private Double totalValue;

    private String status;
}

File: Front-end/layout/src/app/models/rental.ts
import { Car } from "./car";
import { Landlords } from "./landlord";

export class Rental {
    id!: number;
    startDate!: string; 
    returnDate!: string;
    totalValue!: number;
    status!: string;

    car: Car = new Car();
    landlord: Landlords = new Landlords();
}